### PR TITLE
improved docstrings for disjoint_set.jl

### PR DESCRIPTION
**Pull request by AtsushiSakai**
I improved docstrings for disjoint_set.jl

### CODE REVIEW

**Comment by AtsushiSakai** [comment]
@oxinabox could you please review this PR If you have time?

**Comment by oxinabox** [comment]
I was away last week at a company event, and I am away this week at a conference

**Comment by AtsushiSakai** [comment]
@oxinabox Oh OK.  It is ok after next week :  )

File: src/disjoint_set.jl [hunk 1]
```diff
@@ -37,7 +37,9 @@ Get a number of groups
 num_groups(s::IntDisjointSets) = s.ngroups
 Base.eltype(::Type{IntDisjointSets}) = Int
 
-
+# find the root element of the subset that contains x
+# path compression is implemented here
+#
```
Comment (oxinabox) [inline src/disjoint_set.jl]:
```suggestion
```

File: src/disjoint_set.jl [hunk 2]
```diff
@@ -96,15 +124,15 @@ function push!(s::IntDisjointSets)
...
-#
-############################################################
+"""
+    DisjointSets{T}(xs)
 
+A forest of disjoint sets of arbitrary value type T
+
+It is a wrapper of IntDisjointSets, which uses a
+dictionary to map the input value to an internal index
+
```
Comment (oxinabox) [inline src/disjoint_set.jl]:
```suggestion
```

File: src/disjoint_set.jl (hunk 1 above)
Comment (AtsushiSakai) [reply src/disjoint_set.jl]: I fixed it.

File: src/disjoint_set.jl (hunk 2 above)
Comment (AtsushiSakai) [reply src/disjoint_set.jl]: I fixed it.

**Comment by AtsushiSakai** [comment]
@oxinabox Thank you for reviewing. I fixed it based on your review. Please check it again.

**Pull request merged**